Formula/qt.rb

```diff
@@ -23,134 +23,210 @@ class Qt < Formula
+    mirror "https://mirrors.dotsrc.org/qtproject/official_releases/additional_libraries/qtimageformats/6.0/6.0.1/qtimageformats-everywhere-src-6.0.1.tar.xz"
+    mirror "https://mirrors.ocf.berkeley.edu/qt/official_releases/additional_libraries/qtimageformats/6.0/6.0.1/qtimageformats-everywhere-src-6.0.1.tar.xz"
+    sha256 "27a9d6e85dcd56ad981ef2aac27844e782f8cce0598f83283e6fbbd2a3810105"
+  end
 
-  # Find SDK for 11.x macOS
-  # Upstreamed, remove when Qt updates Chromium
-  patch do
-    url "https://raw.githubusercontent.com/Homebrew/formula-patches/92d4cf/qt/5.15.2.diff"
-    sha256 "fa99c7ffb8a510d140c02694a11e6c321930f43797dbf2fe8f2476680db4c2b2"
+  resource "qt3d" do
+    url "https://download.qt.io/official_releases/additional_libraries/qt3d/6.0/6.0.1/qt3d-everywhere-src-6.0.1.tar.xz"
+    mirror "http://mirror.bit.edu.cn/qtproject/official_releases/additional_libraries/qt3d/6.0/6.0.1/qt3d-everywhere-src-6.0.1.tar.xz"
+    mirror "http://ftp.jaist.ac.jp/pub/qtproject/official_releases/additional_libraries/qt3d/6.0/6.0.1/qt3d-everywhere-src-6.0.1.tar.xz"
+    mirror "https://mirrors.dotsrc.org/qtproject/official_releases/additional_libraries/qt3d/6.0/6.0.1/qt3d-everywhere-src-6.0.1.tar.xz"
+    mirror "https://mirrors.ocf.berkeley.edu/qt/official_releases/additional_libraries/qt3d/6.0/6.0.1/qt3d-everywhere-src-6.0.1.tar.xz"
+    sha256 "1234c6904ad13894fb1a798a5540317d241b5cbded70e980e929f477bc060f34"
   end
 
-  # Patch for qmake on ARM
-  # https://codereview.qt-project.org/c/qt/qtbase/+/327649
-  if Hardware::CPU.arm?
-    patch do
-      url "https://raw.githubusercontent.com/Homebrew/formula-patches/9dc732/qt/qt-split-arch.patch"
-      sha256 "36915fde68093af9a147d76f88a4e205b789eec38c0c6f422c21ae1e576d45c0"
-      directory "qtbase"
-    end
+  resource "qtnetworkauth" do
+    url "https://download.qt.io/official_releases/additional_libraries/qtnetworkauth/6.0/6.0.1/qtnetworkauth-everywhere-src-6.0.1.tar.xz"
+    mirror "http://mirror.bit.edu.cn/qtproject/official_releases/additional_libraries/qtnetworkauth/6.0/6.0.1/qtnetworkauth-everywhere-src-6.0.1.tar.xz"
+    mirror "http://ftp.jaist.ac.jp/pub/qtproject/official_releases/additional_libraries/qtnetworkauth/6.0/6.0.1/qtnetworkauth-everywhere-src-6.0.1.tar.xz"
+    mirror "https://mirrors.dotsrc.org/qtproject/official_releases/additional_libraries/qtnetworkauth/6.0/6.0.1/qtnetworkauth-everywhere-src-6.0.1.tar.xz"
+    mirror "https://mirrors.ocf.berkeley.edu/qt/official_releases/additional_libraries/qtnetworkauth/6.0/6.0.1/qtnetworkauth-everywhere-src-6.0.1.tar.xz"
+    sha256 "7f348acedb24e7c33927bd475b87b3c0c9901122f2e11f8c8209de4497e36122"
   end
 
   def install
-    args = %W[
-      -verbose
-      -prefix #{prefix}
+    %w[qtimageformats qt3d qtnetworkauth].each do |addition|
+      resource(addition).stage buildpath/addition
+    end
+
+    # LLVM clobbers the libc++
+    ENV.append "LDFLAGS", "-L /usr/lib"
```

Comment:
I think the correct thing to do here is to use
```ruby
ENV.remove "HOMEBREW_LIBRARY_PATHS", Formula["llvm"].opt_lib
```
so we don't have a runtime dependency on LLVM.

Reply:
llvm is a runtime,dependencies, qdoc uses libclang.dylib and libclang-cpp.dylib, the libclang.dylib not provided by macOS

Reply:
macOS provides `libclang`:
```
❯ l -l /Library/Developer/CommandLineTools/usr/lib
Permissions Size User Date Modified Name
drwxr-xr-x     - root 18 Dec  2020  arc
.r-xr-xr-x  166k root  8 Jan 23:55  arch_tool
drwxr-xr-x     - root 21 Dec  2020  clang
.rwxr-xr-x  197M root  8 Jan 23:54  libclang.dylib
.rwxr-xr-x  171k root  8 Jan 23:54  libcodedirectory.dylib
.rwxr-xr-x  795k root  8 Jan 23:54  libIndexStore.dylib
.rwxr-xr-x   86M root  8 Jan 23:54  libLTO.dylib
.rwxr-xr-x  877k root  8 Jan 23:54  libswiftDemangle.dylib
.rwxr-xr-x  908k root  8 Jan 23:54  libtapi.dylib
.rwxr-xr-x  339k root  8 Jan 23:55  libxcrun.dylib
drwxr-xr-x     - root 21 Dec  2020  llvm-gcc
drwxr-xr-x     - root  9 Feb 23:11  sourcekitd.framework
drwxr-xr-x     - root  9 Feb 23:11  swift
drwxr-xr-x     - root 21 Dec  2020  tapi
```

Reply:
The dependencies of qdoc:
```
./qdoc:
	/usr/local/opt/llvm/lib/libclang.dylib (compatibility version 1.0.0, current version 11.0.1)
	/usr/local/opt/llvm/lib/libclang-cpp.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/llvm/lib/libLLVM.dylib (compatibility version 1.0.0, current version 11.0.1)
	@rpath/QtCore.framework/Versions/A/QtCore (compatibility version 6.0.0, current version 6.0.1)
	/System/Library/Frameworks/DiskArbitration.framework/Versions/A/DiskArbitration (compatibility version 1.0.0, current version 1.0.0)
	/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit (compatibility version 1.0.0, current version 275.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 904.4.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
```
macOS doesn't provide  libclang-cpp, and from qt's [doc](https://doc.qt.io/qt-6/qdoc-guide-clang.html), they recommend use llvm from homebrew, because macOS provided version doesn't provide cmake related files

Reply:
Also in big sur, because of dylib cache, entire qt will link against the libc++ from llvm, not system

Reply:
Depending on LLVM for `qdoc` seems a bit much -- it doesn't seem like essential functionality at all.

>Also in big sur, because of dylib cache, entire qt will link against the libc++ from llvm, not system

Not if llvm libc++ is never loaded into the cache.

Reply:
ok, I will remove the llvm, then qdoc will not be build.

Reply:
In fact, the sql driver and some picture formats support is also the optional, that means user can build the sql driver plugins, and put it to the prefix/share/qt/plugins/sqldrivers.

Reply:
Great. The existing qt formula doesn't install qdoc, so we don't need to add it now. We also don't need to add the plugins that can be built separately if they're not already included in the original qt5 formula.

Reply:
the origin qt build some picture support plugins,  jpeg, tiff, webp, gif etc... these libraries not appear in qt@5 because it uses the bundled version. And if want to built sql plugins separately, there should be a patch to qmake, otherwise the qmake could not find the plugins.

Reply:
I see. How much bigger does the qt installation become if you build the SQL plugins with qt6? If it doesn't add too much in size we can just include it rather than patching qmake.

Reply:
Every sql plugin is about 130KB. Just myy 2 cents, if deliver all sql plugins seems not friendly to the qt and sql developers. Because Qt's MySQL and ODBC plugin can support MariaDB, Percona Server, libiodbc and unixodbc, these formulae in homebrew are marked conflict.

Reply:
OK, maybe we can keep the SQL plugins you're already building here. However, is it possible to convert some of these dependencies to build-only dependencies?

For example, maybe `mysql` and `postgresql` can be build-only so that you don't need to install them if you want the SQL plugins.

Reply:
But it will not pass the `brew audit`.

Reply:
In addition, the IBase plugin can also be built, the open source version is [fire-bird](https://github.com/FirebirdSQL), but must be installed through `brew cask`, because it need creating new macOS account called `firebird`.

Reply:
No, we'll pass on the IBase plugin. A lot of work has been done to make sure no formulae in homebrew/core depend on any casks.